pkg/storage/unified/resource/client.go

```diff
@@ -64,6 +64,17 @@ func NewResourceClient(conn grpc.ClientConnInterface, cfg *setting.Cfg, features
 	})
 }
 
+func NewAuthlessResourceClient(cc grpc.ClientConnInterface) ResourceClient {
+	return &resourceClient{
```

Comment:
Shall we introduce `newResourceClient(grpc.ClientConnInterface)` method to avoid repeating same `&resourceClient{...}` block in `NewAuthlessResourceClient` and `NewLegacyResourceClient`?